### DISCUSSION TITLE
Reebok FR20Z compatible?

### DISCUSSION THREAD

**Discussion by liquidv** [cagnulein/qdomyos-zwift#2091] [General]
Hi, is this app compatible with the Reebok FR20Z treadmill? My treadmill is arriving Saturday so am hoping to get on top of things and see if I could integrate this app with my treadmill and Zwift. Thank you

**Comment by cagnulein** [comment]
Yes it should be! Let me know if you have any questions

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 1 feb 2024 alle 22:58 liquidv ***@***.***> ha
scritto:

**Reply by liquidv** [reply]
Hi Roberto, got my new treadmill and it looks like your app can find the treadmill but it doesn't display any stats.
Tried resetting everything but it's still the same, and when I try to connect the app to Zwift it doesn't find the app. I followed one of your youtube videos on how to 'QZ and Zwift on the same Android device'.

Any ideas? Thanks

![image](https://github.com/cagnulein/qdomyos-zwift/assets/122752718/8654e824-ba30-49cb-8fc3-435cef1a3668)

**Reply by cagnulein** [reply]
Ok first of all let's try with the last build. Click here https://play.google.com/apps/testing/org.cagnulen.qdomyoszwift

And then update it to the 2.16.40

Let me know if you are experiencing the same issue

In case share me a debug log https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

**Reply by liquidv** [reply]
Brilliant, it works thank you! :)
It's also fixed the treadmill speed difference between the treadmill and Zwift itself, Zwift thinks I was walking when I was running on the treadmill!
Will be sending you a coffee!

**Comment by cagnulein** [comment]
Glad to hear it!
Thank you and have a great weekend!


Il giorno sab 3 feb 2024 alle 17:33 liquidv ***@***.***> ha
scritto:

**Reply by liquidv** [reply]
Is the treadmill meant to automatically incline/decline and increase speed/decrease speed in a workout within Zwift?
I tried a few workouts but it says on the screen to set speed/incline, but the treadmill doesn't automatically do this?

**Comment by cagnulein** [comment]
check this
https://github.com/cagnulein/qdomyos-zwift/wiki/Can-Zwift-control-speed-and-or-inclination-of-my-treadmill%3F


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 3 feb 2024 alle 19:37 liquidv ***@***.***> ha
scritto:

**Reply by liquidv** [reply]
ah ok, it works on free run but not the other, didn't know there was a difference, all good, thanks again!

**Comment by liquidv** [comment]
Hi Robert, does your app work on Windows 10? I'm finding my tablet isn't quite up to the task at running Zwift and Casting it onto a monitor so hoping to use a computer instead. Thanks
I notice somewhere it only states Windows 11 only?

**Reply by cagnulein** [reply]
yes https://github.com/cagnulein/qdomyos-zwift/releases/download/nightly/windows-msvc2019-binary.zip 
all windows version are supported,  i guess XP also :D
This is a 15 minutes trial version, you can unlock it with a patreon membership
Let me know

**Comment by liquidv** [comment]
Is it possible to run it like this?

Treadmill > QZ on android phone (blutooth) > Windows pc with Zwift (Bluetooth)?

I've tried all combinations with Experimental feature (android notification off) and it doesn't seem to work.
It can find QZ on bluetooth pairing within Zwift on windows, but going into a map it says connection error on the top left and the avatar doesn't run. The phone Treadmill > QZ android phone is no problem.

**Comment by cagnulein** [comment]
yes it's the classic setup.
but did you rename the phone to "run" and reboot the phone?
also not all the dongle can handle this on windows. which bluetooth dongle
do you have?
also don't you see the wahoo virtual device from  qz on zwift?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 7 feb 2024 alle 20:57 liquidv ***@***.***> ha
scritto:

**Comment by liquidv** [comment]
I have this: ASUS USB-BT400 - Bluetooth 4.0 USB Adapter (USB 2.0, 2.1, 3.0 Compatible) Black
I just renamed my phone to 'run' and rebooted phone, tested again, doesn't seem to work

Tried all 3 connections but they don't work, bluetooth dongle issue?

![image](https://github.com/cagnulein/qdomyos-zwift/assets/122752718/316bd5d8-474e-428b-a6b1-b66122ddf6fe)

**Reply by cagnulein** [reply]
is the firewall on windows disabled? wahoo tread is the right one and it doesn't use bluetooth

**Reply by liquidv** [reply]
the phone with zdomyos would need to connect to the bluetooth on the pc to connect to zwift right?
I've tried disabling firewall, same, connection failure

![image](https://github.com/cagnulein/qdomyos-zwift/assets/122752718/e29bf0aa-edd5-487f-be8a-b20e0cb8112a)

**Reply by cagnulein** [reply]
no, the phone of qz MUST NOT be connected to the windows . unpair it in case

ok for the firewall, it's a zwift cache issue so. you can try to uninstalling it.

did you also already try on the same android device with the android notification icon enabled on the experimental settings?

**Comment by liquidv** [comment]
how does qz pair with zwift through windows? wifi/bluetooth?
bluetooth isnt' paired with windows, but what i meant is zwift finds qz on the phone via bluetooth?
I will try to uninstall/reinstall zwift and see what happens.

I've tried on 2 different samsung phones, A32 and S23, same result.

**Comment by cagnulein** [comment]
try android to android too.
zwift on pc is seriously bad written. it's mentioned everywhere on the forum

qz talk with zwift bluetooth or wifi. bluetooth with the phone name, wahoo
with wifi

let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 7 feb 2024 alle 21:25 liquidv ***@***.***> ha
scritto:

**Reply by liquidv** [reply]
android > android fails
still reinstalling zwift on windows.....

**Comment by cagnulein** [comment]
android on the same android ? did you try?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 7 feb 2024 alle 21:38 liquidv ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
remember the android notification icon setting

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 7 feb 2024 alle 21:38 cagnulein ***@***.***> ha
scritto:

**Comment by liquidv** [comment]
Ok I tried all combinations and only 'qz & zwift on same android phone' works
android > android don't work
android > windows don't work
android notification icon setting enabled ( i see it at the top of the phone)

**Comment by cagnulein** [comment]
this means that you have something on your wifi that drops udp packages.
it's the only explanation

anyway why don't you use qz directly on your pc? you will avoid all these
issues

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 7 feb 2024 alle 21:51 liquidv ***@***.***> ha
scritto:

**Comment by liquidv** [comment]
Ah just tested windows version, seems to work flawless, I will consider the patreon. Thank you

**Comment by cagnulein** [comment]
Glad it worked!


Il giorno mer 7 feb 2024 alle 22:25 liquidv ***@***.***> ha
scritto:

**Comment by liquidv** [comment]
Hi Again, so I just joined the patreon and put my email address in the general settings like it said, but incline doesn't seem to work on windows 10, and I got this message after:

![image](https://github.com/cagnulein/qdomyos-zwift/assets/122752718/e7348377-8a53-419c-8d01-dd2b33cbb4d8)

Also which settings do I use for the treadmill on windows 10?

![image](https://github.com/cagnulein/qdomyos-zwift/assets/122752718/a2e706c1-ed77-4853-b5fe-6963d25f3c98)


Thanks

**Reply by cagnulein** [reply]
hi i just added the license to your email address. auto inclination via api is not yet supported on windows 10

you have to use the ocr way.

so you need to download the windows version with python support

let me know

**Comment by liquidv** [comment]
sorry what is ocr way?
do you have a link with windows version with python support? Thanks

**Comment by cagnulein** [comment]
this one
https://github.com/cagnulein/qdomyos-zwift/releases/download/nightly/windows-binary.zip

leave the setting as you already have and this build will read the zwift
screen (same pc of qz) and will send the metrics to qz!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 8 feb 2024 alle 19:46 liquidv ***@***.***> ha
scritto:

**Comment by liquidv** [comment]
I just downloaded it and assume I'm only meant to use only the new windows-binary version, and not 'windows-msvc2019-binary' ?
The new windows-binary version can't seem to find my treadmill via bluetooth, but the old windows-msvc2019-binary finds it.

Am I doing something wrong?

**Reply by cagnulein** [reply]
hah ok yes the https://github.com/cagnulein/qdomyos-zwift/releases/download/nightly/windows-msvc2019-binary.zip is ok too 
I thought you use one without the python one

So in case share me a log about this.

Zwift must be running in full screen at 1920x1080 in order to collect ocr info

**Reply by liquidv** [reply]
In that case I will go back to the old windows-msvc2019-binary which finds my treadmill.
yes it runs in 1920x1080
sorry where do I find the logs?

**Comment by cagnulein** [comment]
enable the debug log and the file it's in the same qz folder
let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 8 feb 2024 alle ore 20:14 liquidv ***@***.***>
ha scritto:

**Comment by liquidv** [comment]
Sorry I couldn't find a map where there is an incline straight away, if you know of one please let me know.
Had to run about 6 minutes before it showed incline on the map, but no incline on the treadmill, here is the debug

[debug-Thu_Feb_8_19_18_36_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14213788/debug-Thu_Feb_8_19_18_36_2024.log)

**Reply by cagnulein** [reply]
you can use the france or scotland maps.

the problem seems this one

`Thu Feb 8 19:21:17 2024 1707420077093 Debug: .\bluetooth.cpp void __cdecl bluetooth::debug(const class QString &) "python << OUT None\r\n"`

@victorypoint is the master of this feature but he is on vacation til mid Feb. I will check it but I didn't code myself so I could be slow
I will let you know tomorrow!

**Reply by liquidv** [reply]
Thanks!

**Reply by cagnulein** [reply]
@liquidv can you please try the instructions here https://github.com/cagnulein/qdomyos-zwift/issues/1670#issuecomment-1741050576

**Reply by liquidv** [reply]
Hi Roberto, I followed the link and downloaded the new incline file and replaced, also I read it's not compatible with multiple monitors which I had 4 monitors connected to my computer. I changed Zwift/QZ to main monitor.
It works now so I don't know if that new file fixed the issue, or me running Zwift/QZ on the main monitor, all good now.
If you would still like the file I can upload, it's about 120MB though.

Thanks!

**Comment by smoko1** [comment]
Hi guys!
Is possible to connect with your app model: REEBOK FR30 and use Zwift?
Thanks a lot

**Comment by cagnulein** [comment]
Yes, it works!
You need the beta from the store! Are you on ios or android? did you
already download the app?


Il giorno ven 9 feb 2024 alle 10:03 smoko1 ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
that's great! for sure is using the main monitor! so you can replace the
again the python file since we did a patch in the last days to fix
something on some maps.

Let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 9 feb 2024 alle 10:33 liquidv ***@***.***> ha
scritto:

**Reply by liquidv** [reply]
The old python file? I overwritten it with the new one, I assume I would need to download it again?

**Comment by cagnulein** [comment]
yes :)

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 9 feb 2024 alle 10:37 liquidv ***@***.***> ha
scritto:

**Reply by liquidv** [reply]
Thanks!
